chain/chain/src/chain.rs

```diff
@@ -2875,34 +2983,30 @@ impl<'a> ChainUpdate<'a> {
         })
     }
 
-    /// Applies chunks in separate threads, awaits finishing and processes results
-    fn apply_chunks(
+    /// Applies chunks and processes results
+    fn apply_chunks_and_process_results(
         &mut self,
-        me: &Option<AccountId>,
         block: &Block,
         prev_block: &Block,
-        mode: ApplyChunksMode,
+        work: Vec<Box<dyn FnOnce() -> Result<ApplyChunkResult, Error> + Send + 'static>>,
```

Comment:
The `FnOnce() -> Result` signature worries me. It means that we close over some state to do something side-effectful later. That has two problems:

* The state is effectively invisible -- looking at this, it's clear that *something* is going to be modified, but I can't understand what that would be without looking at the call-site which creates the `work`
* The asynchronosy here means that we might have potential issue with capturegin state at one point in time and looking at it later. 

I think the part that carries mutability here is runtime_adapter, so maybe change this to 

```rust
FnOnce(&dyn RuntimeAdapter) -> Result<ApplyChunkResult, Error> + Send + 'static>
```

?

Reply:
The only real mutability inside is on store, not on RuntimeAdapter. I'll think about your suggestion though

Reply:
I actually think it is best to remove Runtime from SyncJonsActor at all and pass it when needed, this seems more clean to me. What do you think? (although this probably out of scope for this PR)

Reply:
Sounds good to me! As far as I understand, `Runtime` is the main state-holding struct, and it's better if there's a single root of all the state. Ie, two actors, each of which owns an Arc to runtime dosen't feel right, it's better if there's only one actor that owns the thing, and just passes references to (snaphots of) it to other things whch want to do stuff in background.

Reply:
#4898